### PR TITLE
Move font style to `section-button` in accordion

### LINKED ISSUE #2481: Accordion styles missing the correct font when javascript not available
## Description of the issue
If javascript is not available, the accordion component does not have full styles.

Screenshot from the design system:
Screenshot 2021-12-16 at 15 15 27

## Steps to reproduce the issue
Navigat eto the accordion page in the design system and disable javascript. The headings for each section will not be using New Transport.

## Actual vs expected behaviour
All text should be using New Transport

## Environment (where applicable)
- Operating system:
- Browser: Chrome 94
- GOV.UK Frontend Version: 4.0.0

### PR DESCRIPTION
**Pull request by JoshuaLicense**
The accordion heading text is currently not styled when JavaScript (JS) is disabled. Current font styling applies to `govuk-accordion__section-heading-text`, but that element only exists when JS is enabled.

**This MR:**
- Moves the `govuk-font()` macro to `govuk-accordion__section-button`.
- Removes `govuk-font()` from `govuk-accordion__section-toggle` and replaces with the explicit macros `govuk-typography-responsive($size: 19)` & `govuk-typography-weight-regular` - to reduce the font property duplication; `govuk-accordion__section-toggle` will inherit the font properties from `govuk-accordion__section-button`.

Closes #2481 

Happy to make any changes if required.

### CODE REVIEW

**Review by 36degrees — APPROVED** [review]
Thanks very much @JoshuaLicense – these changes look good to me 👍🏻 

This just needs another review from our team before we can get this merged.

**Review by owenatgov — APPROVED** [review]
Looks good ot me too 👍🏻  Thanks @JoshuaLicense! This just needs the changelog conflict resolved and we're good to go.

**Pull request merged**
